x/ibc/core/02-client/types/keys.go

```diff
@@ -9,4 +19,39 @@ const (
 
 	// QuerierRoute is the querier route for IBC client
 	QuerierRoute string = SubModuleName
+
+	// KeyNextClientSequence is the key used to store the next client sequence in
+	// the keeper.
+	KeyNextClientSequence = "nextClientSequence"
 )
+
+// IsValidClientID checks if a clientID is in the format required for parsing client
+// identifier. The client identifier must be in the form: `{client-type}-{N}
+var IsValidClientID = regexp.MustCompile(`^.*[^-]-[0-9]{1,20}$`).MatchString
+
+// FormatClientIdentifier returns the client identifier with the sequence appended.
+func FormatClientIdentifier(clientType string, sequence uint64) string {
+	return fmt.Sprintf("%s-%d", clientType, sequence)
+}
+
+// ParseClientIdentifier parses the client type and sequence from the client identifier.
+func ParseClientIdentifier(clientID string) (string, uint64, error) {
+	if !IsValidClientID(clientID) {
+		return "", 0, sdkerrors.Wrapf(host.ErrInvalidID, "invalid client identifier %s is not in format: `{client-type}-{N}`", clientID)
+	}
+
+	splitStr := strings.Split(clientID, "-")
+	lastIndex := len(splitStr) - 1
+
+	clientType := strings.Join(splitStr[:lastIndex], "-")
+	if strings.TrimSpace(clientType) == "" {
+		return "", 0, sdkerrors.Wrap(host.ErrInvalidID, "client identifier must be in format: `{client-type}-{N}` and client type cannot be blank")
```

Comment:
Untested: add case: `-1`

Reply:
Not sure I follow, I already added a [negative sequence case test](https://github.com/cosmos/cosmos-sdk/pull/8034/files#diff-3a8f11a5d5f686b14a2500013990c9eeeb64a38d94ab7e5e66f2c32ef7be9d16R30)

These lines are just sanity checks. They aren't reachable because of the regex check

Reply:
Changed the structure and found some tests to hit code cov